### DISCUSSION TITLE
Object reference not set to an instance of an object in WritableBitmapRenderHandle.CreateOrUpdateBitmap

### DISCUSSION THREAD

**Discussion by Teizakk** [cefsharp/CefSharp#3453] [Q&A]
Hello everyone,

we encountered an pretty interesting "object reference not set to an instance of an object"-Error. We're sadly not able to reproduce it constantly and I'm not sure if it's any mistake made by us. That's the reason, why I won't open an issue, yet. If you say it's more likely an issue topic, i'll start one.

CefSharp.Wpf Version 86.0.241.0
.NET Framework 4.7.1
Windows 10 x64 in an x86 Program

In some cases WritableBitmapRenderHandle.CreateOrUpdateBitmap in line 122 the bitmap seems to be null.

![Callstack](https://user-images.githubusercontent.com/23083689/111791832-e8c29a80-88c3-11eb-84d0-8c0e89aa1a55.JPG)

I think those are pretty rare cases. Do you have an idea in which cases the bitmap could become null in that part? And do you know how I'd be able to prevent that?

**Comment by amaitland** [comment]
> I'm not sure if it's any mistake made by us

What are you doing with the browser? I've not seen any other reports of this null reference.

> Do you have an idea in which cases the bitmap could become null in that part?

Without an example that reproduces the problem it's hard to say.

I'd suggest you implement your own `IRenderHandler` implementation based on `WritableBitmapRenderHandler` and add some additional logging/debugging.

The `WPF` version supports implementing your own rendering implementation. 

http://cefsharp.github.io/api/88.2.x/html/P_CefSharp_Wpf_ChromiumWebBrowser_RenderHandler.htm
